core/domain/summary_services.py

```diff
@@ -257,19 +257,17 @@ def get_exploration_metadata_dicts(exploration_ids, user):
     filtered_exploration_summaries = []
     for (exploration_summary, exploration_rights) in (
             zip(exploration_summaries, exploration_rights_objects)):
-        if exploration_summary is None or exploration_rights is None:
-            continue
-
-        if exploration_summary.status == (
-                rights_manager.ACTIVITY_STATUS_PRIVATE):
-            if user.user_id is None:
-                continue
+        if exploration_summary is not None and exploration_rights is not None:
```

Comment:
This is harder to read than the original code. What was wrong with the previous one? 

You should really try to construct scenarios that lead to the error cases being triggered, and testing those.

Reply:
I am facing same problems as prod_validation_jobs. It seems coverage.py is not counting those statements. I tried raising an exception('xxxxx') in place of 'continue' and that exception gets raised which means it gets called but the continue statement is not showing as covered

Reply:
Seems related to https://github.com/nedbat/coveragepy/issues/198. OK, we'll leave it as you have it, then. Thanks for the explanation.